lint/failure.go

```diff
@@ -5,6 +5,30 @@ import (
 	"go/token"
 )
 
+const (
+	FailureCategoryArgOrder            = "arg-order"
+	FailureCategoryBadPractice         = "bad practice"
+	FailureCategoryCodeStyle           = "code-style"
+	FailureCategoryComments            = "comments"
+	FailureCategoryComplexity          = "complexity"
+	FailureCategoryContent             = "content"
+	FailureCategoryErrors              = "errors"
+	FailureCategoryImports             = "imports"
+	FailureCategoryLogic               = "logic"
+	FailureCategoryMaintenance         = "maintenance"
+	FailureCategoryNaming              = "naming"
+	FailureCategoryOptimization        = "optimization"
+	FailureCategoryStyle               = "style"
+	FailureCategoryTime                = "time"
+	FailureCategoryTypeInference       = "type-inference"
+	FailureCategoryUnaryOp             = "unary-op"
+	FailureCategoryUnexportedTypeInAPI = "unexported-type-in-api"
+	FailureCategoryZeroValue           = "zero-value"
+
+	failureCategoryInternal = "REVIVE_INTERNAL"
+	failureCategoryValidity = "validity"
+)
```

Comment:
Most of them are exported. Do we need it?

Because of the modules and packages we do. But do we need them to be exported outside the module? If not, maybe it could be a internal/failure packages, allowing the constants to be unexported outside the module

Reply:
If someone wants to create a custom rule (not included in Revive), these might be useful.
I'd however make them typed, to avoid further direct strings used for this purpose.

Reply:
OK for me then.

And yes, good idea about the typed constants